tests/func/test_commit.py

```diff
@@ -1,86 +1,82 @@
+import pytest
 from dvc.utils.stage import load_stage_file, dump_stage_file
 
-from tests.basic_env import TestDvc
 from dvc.stage import StageCommitError
 
 
-class TestCommitRecursive(TestDvc):
-    def test(self):
-        stages = self.dvc.add(self.DATA_DIR, recursive=True, no_commit=True)
-        self.assertEqual(len(stages), 2)
+def test_commit_recursive(dvc, repo_dir):
```

Comment:
Btw, could you please check that dvc.root_dir and repo_dir._root_dir are the same? I'm a bit worried about the order of fixture execution.

Reply:
I can, but I think we can rely on pytest to make sure we use the same instances of fixture. (I did manual id check, just to be sure and it seems to work that way).

Here is the reference
https://docs.pytest.org/en/latest/fixture.html#automatic-grouping-of-tests-by-fixture-instances

```
pytest minimizes the number of active fixtures during test runs. If you have a parametrized fixture, then all the tests using it will first execute with one instance and then finalizers are called before the next fixture instance is created
```
Althought this example is to show parametrized fixtures, I think it also applies to our use case.
New fixture (repo_dir) will not be created before finalizing the old one, which will not be finalized until we stop using the dvc fixture (basically until the end of the test). 

Also in

https://docs.pytest.org/en/latest/fixture.html#scope-sharing-a-fixture-instance-across-tests-in-a-class-module-or-session

It is mentioned that:
```
Extending the previous example, we can add a scope="module" parameter to the @pytest.fixture invocation to cause the decorated smtp_connection fixture function to only be invoked once per test module (the default is to invoke once per test function)
```
Which would also suggest that pytest reuses fixture instead calling it again.

Reply:
Thanks! :slightly_smiling_face: